### PR TITLE
enhance(templates): add toYaml function

### PR DESCRIPTION
**Pull request by JordanSussman**
Adding the custom Golang template function of `toYaml` that was copied from [github.com/helm/helm](https://github.com/helm/helm/blob/a499b4b179307c267bdf3ec49b880e3dbd2a5591/pkg/engine/funcs.go#L52). This new function makes it easy to directly consume variables within the underlying template as yaml.

This functionality was originally discussed within [slack](https://gophers.slack.com/archives/CNRRKE8KY/p1635286594017500).

### CODE REVIEW

**Review by kneal — DISMISSED** [review]
LGTM 🐬

**Review by jbrockopp — DISMISSED** [review]
LGTM

File: compiler/template/native/convert.go [hunk 1]
```diff
@@ -23,6 +25,22 @@ func convertPlatformVars(slice raw.StringSliceMap, name string) raw.StringSliceM
 	return envs
 }
 
+// toYAML takes an interface, marshals it to yaml, and returns a string. It will
+// always return a string, even on marshal error (empty string).
+//
+// This code is under copyright (full attribution in NOTICE) and is from:
+// https://github.com/helm/helm/blob/a499b4b179307c267bdf3ec49b880e3dbd2a5591/pkg/engine/funcs.go#L83
```
Comment (wass3r) [inline compiler/template/native/convert.go:32]: looks like we need a nolint here :(

**Review by wass3r — CHANGES_REQUESTED** [review]

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by wass3r — APPROVED** [review]

**Pull request merged**